src/modules/collection/sagas.spec.ts

```diff
@@ -2377,3 +2380,163 @@ describe('when handling a deploy missing entities', () => {
     })
   })
 })
+
+describe('when publishing a collection with credits', () => {
+  let collection: Collection
+  let items: Item[]
+  let email: string
+  let subscribeToNewsletter: boolean
+  let from: string
+  let creditsResponse: CreditsResponse
+  let txHash: string
+  let creditsServerUrl: string
+
+  beforeEach(() => {
+    collection = getCollectionMock({
+      salt: '0x' + '123'.padStart(64, '0'),
+      id: 'someId',
+      name: 'name',
+      urn: 'urn:decentraland:amoy:collections-v2:0xabc123:collection-name',
+      isPublished: false
+    })
+    items = [
+      getItemMock(collection, {
+        collectionId: 'someId',
+        currentContentHash: 'bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi',
+        blockchainContentHash: 'bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi',
+        catalystContentHash: 'bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi',
+        contents: { 'thumbnail.png': 'bafkreiaxnnnb7qz2focittuqq3ya25q7rcv3bqynnczfzfqnugriqvcbey' }, // New hash format, not starting with 'Qm'
+        rarity: 'common' as any, // Added rarity to prevent toLowerCase error
+        price: '100000000000000000000' // 100 MANA
+      })
+    ]
+    email = 'test@example.com'
+    subscribeToNewsletter = false
+    from = '0x' + '123'.padStart(40, '0')
+    txHash = '0xdeadbeef'
+    creditsServerUrl = 'https://credits.example.com'
+    creditsResponse = {
+      credits: [
+        {
+          id: 'credit-1',
+          amount: '1000000000000000000', // 1 MANA
+          availableAmount: '1000000000000000000',
+          expiresAt: '9999999999',
+          signature: '0xsignature',
+          contract: '0x5309ae874fc4eb21adcd63f8b6c3f766cc3b1849',
+          season: 1,
+          timestamp: '1234567890',
+          userAddress: from
+        }
+      ],
+      totalCredits: 1000000000000000000
+    }
+  })
+
+  describe('when using credits with MANA payment method', () => {
+    let originalFetch: typeof global.fetch
+    let useCreditsCollectionManagerSpy: jest.SpyInstance
+
+    beforeEach(() => {
+      ;(mockBuilder.fetchCollectionItems as jest.Mock).mockResolvedValue(items)
+      // Mock fetch globally for credits server signature
+      originalFetch = global.fetch
+      global.fetch = jest.fn().mockResolvedValue({
+        ok: true,
+        json: () => Promise.resolve({ signature: '0xexternalCallSignature', messageHash: '0xhash' } as any)
+      } as any)
```

Comment:
If we pass the `creditsService` to the sagas, we shouldn't need to mock the fetch globally, which would simplify our tests.